### PR TITLE
add Redis cache 

### PR DESCRIPTION
**Pull request by Nedzeya**
add cache into  getAvailableLocations method class LocationService, 
add test caching in integration test with duration printing in console

### CODE REVIEW

**Review by Nedzeya — COMMENTED** [review]
I made a lot of code changes because I applied the xCloud style.

File: src/main/java/com/klachkova/locationsystem/services/LocationService.java [hunk 1]
```diff
@@ -54,6 +59,7 @@ public LocationService(UserRepository userRepository,
      * <p>
      * Converts the provided LocationDTO to a Location entity, validates it, and saves it to the repository.
      * Associates the location with the user specified in the DTO.
+     * Caches the registered location for future retrieval.
```
Comment (RSlabodchikov) [inline src/main/java/com/klachkova/locationsystem/services/LocationService.java]: I think this method would not update cache

File: src/main/java/com/klachkova/locationsystem/services/LocationService.java [hunk 2]
```diff
@@ -106,23 +115,25 @@ private void validateLocation(Location location) {
...
 
     /**
-     * Retrieves a location by its ID.
-     * <p>
-     * Throws an exception if no location with the given ID is found.
-     * </p>
+     * Retrieves a Location by its ID.
+     * First checks the Redis cache for the location.
+     * If the location is not found in the cache, it retrieves
+     * the location from the database and caches it for future access.
```
Comment (RSlabodchikov) [inline src/main/java/com/klachkova/locationsystem/services/LocationService.java]: Please check if its correct place for this comment, probably its related to `etAvailableLocations`?

File: src/main/java/com/klachkova/locationsystem/util/config/RedisConfig.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+@Configuration
+@EnableCaching
+public class RedisConfig {
+
+    @Bean
+    public RedisCacheManager cacheManager(RedisConnectionFactory redisConnectionFactory) {
+
+        RedisCacheConfiguration cacheConfiguration = RedisCacheConfiguration.defaultCacheConfig()
+            .entryTtl(Duration.ofMinutes(10))
```
Comment (RSlabodchikov) [inline src/main/java/com/klachkova/locationsystem/util/config/RedisConfig.java]: Lets make TTL configurable (via application properties f.e.)

**Review by RSlabodchikov — APPROVED** [review]

File: src/main/java/com/klachkova/locationsystem/services/LocationService.java (hunk 2 above)
Comment (Nedzeya) [reply src/main/java/com/klachkova/locationsystem/services/LocationService.java]: I initially implemented the cache for these methods too, then I deleted the changes to the methods and forgot about the documentation.
Edited

**Pull request merged**